### PR TITLE
Update lockfile (& Docker)

### PR DESCRIPTION
**Pull request by nhynes**
re: https://github.com/oasislabs/oasis-parity/pull/192

This also brings the docker image size down from 70M to 18M.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,31 +1,14 @@
...
- RUSTFLAGS='-C target-feature=+aes,+ssse3' /usr/local/lib/cargo/bin/cargo install --locked --git https://github.com/oasislabs/oasis-chain.git oasis-chain && \
- cp /usr/local/lib/cargo/bin/oasis-chain /usr/local/bin/oasis-chain && \
- /usr/local/lib/cargo/bin/rustup self uninstall -y && \
- apt-get remove --purge --yes curl ca-certificates git build-essential && \
- apt autoremove --yes && \
- apt-get clean && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+RUN cargo build --release --locked
 
-FROM ubuntu:bionic as oasis-chain-runner
+FROM gcr.io/distroless/cc
```
Comment (mitar) [inline Dockerfile:8]: Nice. :-)

Under which Linux user (uid) does the distroless run stuff?

File: Dockerfile [hunk 2]
```diff
@@ -1,31 +1,14 @@
...
 
-RUN \
- apt-get update -q -q && \
- apt-get install --yes tzdata curl ca-certificates git build-essential && \
- echo 'UTC' > /etc/timezone && \
- rm /etc/localtime && \
- dpkg-reconfigure tzdata && \
- apt-get upgrade --yes && \
- curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --no-modify-path --default-toolchain nightly && \
- RUSTFLAGS='-C target-feature=+aes,+ssse3' /usr/local/lib/cargo/bin/cargo install --locked --git https://github.com/oasislabs/oasis-chain.git oasis-chain && \
```
Comment (mitar) [inline Dockerfile:15]: `RUSTFLAGS='-C target-feature=+aes,+ssse3'` flags are not needed anymore?

File: .dockerignore [hunk 3]
```diff
@@ -1,5 +1,4 @@
 **/.gitignore
-.*
```
Comment (mitar) [inline .dockerignore:2]: Why removing this?

Comment (nhynes) [reply .dockerignore:2]: * to pull in the rustflags from the [.cargo/config](https://github.com/oasislabs/oasis-chain/blob/master/.cargo/config#L2)
* to get the git rev for the [build script](https://github.com/oasislabs/oasis-chain/blob/master/build.rs)

File: Dockerfile (hunk 1 above)
Comment (nhynes) [reply Dockerfile:8]: yeah, great tip. thanks!

> Under which Linux user
root, it seems. they also create a `nobody`, so I've modified the dockerfile to use that

Comment (mitar) [reply Dockerfile:8]: Awesome.

**Pull request merged**